modules/rtm/tests/testthat/test.resample.R

```diff
@@ -0,0 +1,51 @@
+library(testthat)
+library(PEcAnRTM)
+context("Resampling functions")
+
+test_that(
+  "Univariate resampling works",
+  {
+    from <- seq(2, 20, 2)
+    to <- seq(2, 20)
+    values <- from ^ 2
+    true_values <- to ^ 2
+    resample_values <- resample(values, from, to)
+    expect_equal(true_values, resample_values)
+  }
+)
+
+test_that(
+  "Multivariate resampling works",
+  {
+    m <- matrix(1:20, ncol = 2)
+    true_m <- cbind(seq(1, 10, 0.5), seq(11, 20, 0.5))
+    resample_m <- resample(m, 1:10, seq(1, 10, 0.5))
+    expect_equal(true_m, resample_m)
+  }
+)
+
+data(testspec)
+test_that(
+  "Spectra resampling works",
+  {
+    spec <- spectra(testspec_ACRU[, 1:5], 400:2500)
+    new_wl <- seq(400, 1300, 10)
```

Comment:
Can we add a test to convert from "coarse" to 1nm or finer spec as well?   Or I could just test that myself by reading in some AVIRIS or g-liht spec.

Reply:
Yeah, this is probably a good idea. I'll just need to figure out a good error tolerance for `expect_equal`.